man/install-info.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.49.1.
.TH INSTALL-INFO "1" "September 2024" "GNU texinfo 7.1.1" "User Commands"
.SH NAME
install-info \- update info/dir entries
.SH SYNOPSIS
.B install-info
[\fI\,OPTION\/\fR]... [\fI\,INFO-FILE \/\fR[\fI\,DIR-FILE\/\fR]]
.SH DESCRIPTION
Add or remove entries in INFO\-FILE from the Info directory DIR\-FILE.
INFO\-FILE and DIR\-FILE are required unless the \fB\-\-info\-file\fR
or \fB\-\-dir\-file\fR (or \fB\-\-info\-dir\fR) options are given, respectively.
.SH OPTIONS
.TP
\fB\-\-add\-once\fR
add only to first matching section, not all.
.TP
\fB\-\-align\fR=\fI\,COL\/\fR
start description of new entries at column COL.
.TP
\fB\-\-calign\fR=\fI\,COL\/\fR
format second and subsequent description lines to
start at column COL.
.TP
\fB\-\-debug\fR
report what is being done.
.TP
\fB\-\-delete\fR
delete existing entries for INFO\-FILE from DIR\-FILE;
don't insert any new entries.
.TP
\fB\-\-defsection\fR=\fI\,TEXT\/\fR
like \fB\-\-section\fR, but only use TEXT if no sections
are present in INFO\-FILE (replacing "Miscellaneous").
.TP
\fB\-\-description\fR=\fI\,TEXT\/\fR
the description of the entry is TEXT; used with
the \fB\-\-name\fR option to become synonymous with the
\fB\-\-entry\fR option.
.TP
\fB\-\-dir\-file\fR=\fI\,NAME\/\fR
specify file name of Info directory file;
equivalent to using the DIR\-FILE argument.
.TP
\fB\-\-dry\-run\fR
same as \fB\-\-test\fR.
.TP
\fB\-\-entry\fR=\fI\,TEXT\/\fR
insert TEXT as an Info directory entry,
overriding any corresponding entry from DIR\-FILE.
TEXT is written as an Info menu item line followed
.TP
by zero or more extra lines starting with whitespace.
If you specify more than one entry, all are added.
If you don't specify any entries, they are determined
.IP
from information in the Info file itself.
.TP
\fB\-\-help\fR
display this help and exit.
.TP
\fB\-\-info\-dir\fR=\fI\,DIR\/\fR
same as \fB\-\-dir\-file\fR=\fI\,DIR\/\fR/dir.
.TP
\fB\-\-info\-file\fR=\fI\,FILE\/\fR
specify Info file to install in the directory;
equivalent to using the INFO\-FILE argument.
.TP
\fB\-\-item\fR=\fI\,TEXT\/\fR
same as \fB\-\-entry\fR=\fI\,TEXT\/\fR.
.TP
\fB\-\-keep\-old\fR
do not replace entries, or remove empty sections.
.TP
\fB\-\-maxwidth\fR, \fB\-\-max\-width\fR=\fI\,COL\/\fR
wrap description at column COL.
.TP
\fB\-\-menuentry\fR=\fI\,TEXT\/\fR
same as \fB\-\-name\fR=\fI\,TEXT\/\fR.
.TP
\fB\-\-name\fR=\fI\,TEXT\/\fR
the name of the entry is TEXT; used with \fB\-\-description\fR
to become synonymous with the \fB\-\-entry\fR option.
.TP
\fB\-\-no\-indent\fR
do not format new entries in the DIR file.
.TP
\fB\-\-quiet\fR
suppress warnings.
.TP
\fB\-\-regex\fR=\fI\,R\/\fR
put this file's entries in all sections that match the
regular expression R (ignoring case).
.TP
\fB\-\-remove\fR
same as \fB\-\-delete\fR.
.TP
\fB\-\-remove\-exactly\fR
only remove if the info file name matches exactly;
suffixes such as .info and .gz are not ignored.
.TP
\fB\-\-section\fR=\fI\,SEC\/\fR
put entries in section SEC of the directory.
If you specify more than one section, all the entries
.TP
are added in each of the sections.
If you don't specify any sections, they are determined
.TP
from information in the Info file itself;
if nothing is available there, the \fB\-\-defsection\fR
value is used; if that is not specified, the
final default is "Miscellaneous".
.TP
\fB\-\-section\fR R SEC
equivalent to \fB\-\-regex\fR=\fI\,R\/\fR \fB\-\-section\fR=\fI\,SEC\/\fR \fB\-\-add\-once\fR.
.TP
\fB\-\-silent\fR
suppress warnings.
.TP
\fB\-\-test\fR
suppress updating of DIR\-FILE.
.TP
\fB\-\-version\fR
display version information and exit.
.SH "REPORTING BUGS"
Email bug reports to bug\-texinfo@gnu.org,
general questions and discussion to help\-texinfo@gnu.org.
.br
Texinfo home page: http://www.gnu.org/software/texinfo/
.SH COPYRIGHT
Copyright \(co 2024 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
.br
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
.SH "SEE ALSO"
The full documentation for
.B install-info
is maintained as a Texinfo manual.  If the
.B info
and
.B install-info
programs are properly installed at your site, the command
.IP
.B info install-info
.PP
should give you access to the complete manual.


NEWS
This NEWS file records noteworthy changes, very tersely.
See the manual for detailed information.

  Copyright 1992-2023 Free Software Foundation, Inc.

  Copying and distribution of this file, with or without modification,
  are permitted in any medium without royalty provided the copyright
  notice and this notice are preserved.

------------------------------------------------------------------------------

7.1.1 (7 September 2024)
This is a bug-fix release with minimal changes.

* texi2any
  . fix potential crash when @include is used inside a table
  . do not complain about presence of @anchor inside @item in a table
  . C source files that are generated from *.xs files are no no longer
    distributed, so xsubpp from Perl is needed to build XS modules.
  . fix bug that led to memory alignment error on SPARC
  . performance improvement and fixes for MinGW
  . test failures due to floating point rounding error fixed (observed
    on IBM POWER9 processor)

* info
  . crash when setting style to invalid value fixed
  . potential call of memcpy with null argument fixed


7.1 (18 October 2023)
* Language
 . new generic definition commands, @defblock, @defline and @deftypeline,
   for definitions without automatic index entries
 . new @linemacro facility eases use of generic definition commands
 . new command @link creates plain links (supported output formats only)
 . @cartouche takes an argument to specify the cartouche title
 . you can use the new commands @nodedescription and @nodedescriptionblock
   to give text to be used in menu descriptions in Info and HTML output

* texi2any
 . @itemx at the beginning of a @table is now an error, not a warning
 . better validity checking of deeply nested commands
 . check that @set and @clear only appear at the start of a line
 . warn about missing menu entries even if CHECK_NORMAL_MENU_STRUCTURE is
   not set.  you can turn this off by setting CHECK_MISSING_MENU_ENTRY to 0.
 . no longer use --enable-encoding and --disable-encoding to determine
   whether to output encoded characters (instead of entities or commands)
   for HTML, XML, DocBook and LaTeX; instead, use the value of the
   OUTPUT_CHARACTERS customization variable.
 . stricter checks on input encoding, in particular more warnings and
   errors with malformed UTF-8
 . support any input file encoding if support exists in the operating
   system, not just a selected list of encodings
 . resolve an alias referring to another alias at definition time
 . internally, use "source marks" to keep all Texinfo source information that
   is not in the final tree (location of macros, values and included files
   expansion, @if* blocks, DEL comment, and @ protecting end of line on @def*
   lines)
 . HTML output:
    . format @subentry and index entries with @seealso or @seeentry in a more
      similar way to printed output
    . output @shortcontents before @contents by default
    . omit colons after index entries by default.  this can still be
      configured with INDEX_ENTRY_COLON.
    . add @example syntax highlighting as a texi2any extension
    . no more capitalization of @sc argument in HTML Cross-references
    . change @point expansion to U+22C6 in HTML Cross-references
    . if a @node is not associated with a sectioning command but is
      followed by a heading command not usually associated to nodes
      such as @heading and this command appears before other formatted
      content, the heading command is assumed to supply the node heading.
      you can customize this with USE_NEXT_HEADING_FOR_LONE_NODE.
 . Info output:
    . new variable ASCII_DASHES_AND_QUOTES, on by default,
      outputs ASCII characters for literal quote or hyphen characters
      in source, rather than UTF-8.  this makes it easier to search
      Info files.
    . new ASCII_GLYPH variable for using ASCII renditions for glyph
      commands (like @bullet)
    . ASCII_PUNCTUATION still includes the effect of these new variables.
    . new variables AUTO_MENU_DESCRIPTION_ALIGN_COLUMN and AUTO_MENU_MAX_WIDTH
      control the format of descriptions in generated menus
 . XML output:
    . place menu leading text and menu separators in elements instead
      of attributes

* texi2dvi
 . macro expansion with texi2any requires at least version 5.0 (only
   happens with --expand option or with very old texinfo.tex)

* texinfo.tex
 . in @code, ` and ' output by default with backtick and undirected
   single quote glyphs in the typewriter font.  you can still configure
   this using the @codequoteundirected/@codequotebacktick commands.
 . do not insert a space for @ def line continuation, matching the behavior
   of texi2any
 . align section titles in table of contents when more than 10 sections
 . microtype is off by default, for speed
 . page headings generation is no longer linked to the @titlepage command

* info
 . when going Up, position cursor on menu entry for current node
 . allow mouse scrolling support regardless of termcap entries.  this
   supports some more xterm configurations.
 . do not use "/index" as a possible file extension for Info files

* Distribution
  . autoconf 2.71, automake 1.16.5, gettext 0.21



7.0.3 (26 March 2023)
This is a bug-fix release with minimal changes.

* texi2any
  . fix performance regression when Perl binary extension (XS) modules
    are not being used (e.g. with TEXINFO_XS=omit)

* info
  . further fix of recoding of UTF-8 files to ASCII to avoid text
    disappearing from nodes
  . avoid possible freeze at start of a file with `-v nodeline=pointers'


7.0.2 (22 January 2023)
This is a bug-fix release with minimal changes.

* texi2any
  . do not distribute architecture-dependent files
  . build fixed on OpenIndiana 11

* info
  . further fix of recoding of UTF-8 files to ASCII
  . fix check for presence of man pages on Solaris

* install-info
  . fix build by avoiding function name clash on some platforms
  . compiler warning re strncat silenced




debian/install-info.sh
#!/bin/sh
#
# Wrapper to the GNU's install-info, to be compatible with the one that used to
# be packaged by dpkg on Debian.
#
# written by Norbert Preining, this is not copyrightable ;-)
#
set -e

if [ -z "$DPKG_RUNNING_VERSION" ] ; then
  # it seems we are running from outside a maintainer script, so give a
  # warning and call ginstall-info without anything else
  # we are so long in the transition, we don't need to tell this anymore!
  # echo "This is not dpkg install-info anymore, but GNU install-info" >&2
  # echo "See the man page for ginstall-info for command line arguments" >&2
  ginstall-info "$@"
else 
  # we are running from a maintainer script, simply ignore the call
  # since we have trigger support and people should rebuild their
  # package with new debhelper which does not add calls to install-info
  # Do not complain if called with "--remove" or "--remove-exactly",
  # as these are used in old packages' prerm scripts (see #546165)
  while [ -n "$1" ]; do
    case "$1" in
      --remove|--remove-exactly)
        exit 0
        ;;
      *)
        shift
        ;;
    esac
  done
  echo "Ignoring install-info called from maintainer script" >&2
  echo "The package $DPKG_MAINTSCRIPT_PACKAGE should be rebuilt with new debhelper to get trigger support" >&2
fi


